### ISSUE TITLE
Implement proof key validation

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/richdocuments#1652]
https://wopi.readthedocs.io/en/latest/scenarios/proofkeys.html

**Comment by drupol** [comment]
Hi,

I build a library that is able to validate WOPI client request, you can find the library here: https://github.com/Champs-Libres/wopi-lib and the relevant key validator service here: https://github.com/Champs-Libres/wopi-lib/blob/master/src/Service/WopiProofValidator.php

I also created [a bundle for Symfony](https://github.com/Champs-Libres/wopi-bundle) and in order to successfully run the [WopiValidator](https://github.com/microsoft/wopi-validator-core) tests (which doesn't contains `X-WOPI-Proof` and `X-WOPI-ProofOld` headers), I had to create a `RelaxWopiProofValidator` service which only validate the request only and only if the request has the relevant headers: https://github.com/Champs-Libres/wopi-test-bundle/blob/master/src/Service/RelaxWopiProofValidator.php

You can test all of this in a Symfony app that I created here: https://wopi-app.herokuapp.com/

I tried to get in touch with NextCloud in order to discuss about that, but so far I got no reply yet. I guess it's just a matter of time.

Let me know what you think.

**Comment by juliushaertl** [comment]
> I tried to get in touch with NextCloud in order to discuss about that, but so far I got no reply yet. I guess it's just a matter of time.

Yes, sorry I had a bit of a busy week. I'll get back to you soon by mail as well :)

> I build a library that is able to validate WOPI client request, you can find the library here: Champs-Libres/wopi-lib and the relevant key validator service here: https://github.com/Champs-Libres/wopi-lib/blob/master/src/Service/WopiProofValidator.php

Your library is definitely something that I'd consider integrating as it would help us to clean up the WOPI implementation and improve the overall structure of our codebase in that regard. The only thing that would be blocking from my perspective, is that the Nextcloud caching layer currently does not stick to PSR16.

**Comment by drupol** [comment]
Hi Julius,

No worries about the delay, what matter is that you got my message, now for the rest it can wait, there's no hurry.

Regarding WOPI-Lib and PSR16, don't worry about that, we can make an adapter that would act as a bridge, so, there's nothing to be afraid of, and I can help for that for sure, I actually would be glad to help.

**Comment by juliusknorr** [comment]
Implemented in https://github.com/nextcloud/richdocuments/pull/4902

**Comment by drupol** [comment]
Hey! 

Long time no see 😅

Have you had the opportunity to check out the wopi library in the end ? 

The encryption decryption part was made in it already iirc .
